File: quivi.pos.react.app/src/hooks/api/Dtos/orders/Order.ts
import { SessionItem } from "../sessions/SessionItem";
import { OrderState } from "./OrderState";

export interface OrderFieldValue {
    readonly id: string;
    readonly value: string;
}

export interface Order {
    readonly id: string;
    readonly channelId: string;
    readonly sequenceNumber: string;
    readonly state: OrderState;
    readonly employeeId?: string;
    readonly isTakeAway: boolean;
    readonly items: SessionItem[];
    readonly fields: OrderFieldValue[];
    readonly scheduledTo?: string;
    readonly createdDate: string;
    readonly lastModified: string;
}

File: Quivi.Pos.Api/Dtos/Order.cs
using Quivi.Domain.Entities.Pos;

namespace Quivi.Pos.Api.Dtos
{
    public class Order
    {
        public required string Id { get; init; }
        public required string SequenceNumber { get; init; }
        public required string ChannelId { get; init; }
        public string? EmployeeId { get; init; }
        public required IEnumerable<SessionItem> Items { get; init; }
        public required IEnumerable<OrderFieldValue> Fields { get; init; }
        public OrderState State { get; init; }
        public OrderOrigin OrderOrigin { get; init; }
        public bool IsTakeAway { get; init; }
        public DateTimeOffset CreatedDate { get; init; }
        public DateTimeOffset LastModified { get; init; }
    }

    public class OrderFieldValue
    {
        public required string Id { get; init; }
        public required string Value { get; init; }
    }
}